### PR TITLE
Actuator /configprops shows Duration properties as { "units": [ "SECONDS", "NANOS" ] }

### LINKED ISSUE #16526: Actuator /configprops shows Duration properties as { "units": [ "SECONDS", "NANOS" ] }
Spring Boot 2.1.4.RELEASE.

Actuator `/configprops` endpoint shows this property class:

```java
@ConfigurationProperties("demo.props")
@Data
public class DemoProperties {
    private Duration duration = Duration.ofMinutes(10);
}
```

like so:

```js
{
  "contexts": {
    "application": {
      "beans": {
        "demo.props-com.example.demo.DemoProperties": {
          "prefix": "demo.props",
          "properties": {
            "duration": {
              "units": [
                "SECONDS",
                "NANOS"
              ]
            }
          }
        }
      }
    }
  }
}
```

The actual value is not show in any way. I would expect the actual value in the ISO-8601 format or the Spring Boot "human-readable" format.

### PR DESCRIPTION
**Pull request by dreis2211**
Hi,

this is an attempt to fix #16526 .

Feel free to decline the PR if you had a different solution in mind.
Cheers,
Christoph

### CODE REVIEW

**Comment by philwebb** [comment]
Wow that was quick! Thanks @dreis2211 !

**Comment by dreis2211** [comment]
Let me fix the checkstyle issue. Sorry

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java [hunk 1]
```diff
@@ -171,6 +176,7 @@ private ConfigurationBeanFactoryMetadata getBeanFactoryMetadata(
 	 */
 	protected void configureObjectMapper(ObjectMapper mapper) {
 		mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+		mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: Is that related to the support of `Duration`?

Comment (dreis2211) [reply spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: Yes and no - this depends on how you want the durations to be displayed. See https://github.com/FasterXML/jackson-datatype-jsr310/blob/master/src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/DurationSerializer.java#L63

Comment (dreis2211) [reply spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java]: @snicoll I pushed a revised version that registers a module. I've tried this actually before, but it didn't work because I called it before the serializerFactory is overwritten again in `applySerializationModifier`. Anyway, this should be easier now. And btw: there would be a way of calling `ObjectMapper.findAndRegisterModules` but that might register more than you want and is sort of expensive.

**Comment by snicoll** [comment]
While trying to backport this to `2.1.x`, I've noticed that `JavaTimeModule` is not available so I've changed something to include it only when the relevant module is on the classpath. It works on `master` because the new version of `neo4j-ogm-api` has a compile dependency on it. 

I do wonder though if that's what we want to do for the endpoint. Given the endpoint is focused on `@ConfigurationProperties` and we have a dedicated format for `Duration`, Maybe we should use our converter rather than the standard `Duration` format.

Flagging for team attention to see what the rest of the team thinks.

**Comment by wilkinsona** [comment]
I don't like the idea of the actuator's output format changing depending on what's on the classpath. If we need the `JavaTimeModule` to always be available to produce predictable output then I think we should make it a runtime dependency of the module.

The format for a duration in `application.properties` is, from a user's perspective, intended only to be human readable and writable. The format in the Actuator's response is intended to both human and machine readable. As such, I think we should use a standard format for the duration, rather than our own. Assuming I've read the code correctly, this proposal will result in an ISO-8601 seconds based representation which sounds good to me.

**Comment by snicoll** [comment]
Thanks for the feedback and good point on machine readable. Given the side effect of adding the module, I am moving this fix to `2.2.x`.

**Comment by dreis2211** [comment]
Anything I can do to help here, @snicoll ?

**Comment by snicoll** [comment]
@dreis2211 if you want to push force an update that adds the jackson module in runtime scope, feel free to do it but I am more than happy to do that as part of the polish later today.

**Comment by dreis2211** [comment]
I would be only available this evening, so feel free to do it. Highly appreciated - as usual :)

**Comment by snicoll** [comment]
Thanks again @dreis2211!
